(doc) Add info about Maintenance and Support (#450)
chocolatey/docs: input/shared/maintenance-and-support.txt

## Maintenance and Support Contract Considerations for Perpetual License Customers

Having a Maintenance and Support contract with Chocolatey Software, Inc. consists of two parts:

- Maintenance - allows you to download newer versions of the licensed components of Chocolatey.
- Support - allows you to submit issue tickets to our Helpdesk and also have access to our Support Team.

During the active period of the contract, you have access to both `Maintenance` and `Support`.
Once that expires, you would simply need to purchase that contract again to continue access to newer versions of those components.

A Perpetual License means that you have a license which never expires and you are able to use the Chocolatey Licensed Extension with versions of Chocolatey CLI that are compatible.

> While your Maintenance and Support contract is valid, you are able to download updates to the licensed components.

Reviewer: ```suggestion
While your Maintenance contract is valid, you are able to download updates to the licensed components.
```

Reviewer: This part only requires a maintenance contract and not a support contract.

> A Perpetual License customer that does not have an active Maintenance and Support contract will not be in a position to upgrade to the latest version of Chocolatey Licensed Extension.

Reviewer: ```suggestion
A Perpetual License customer that does not have an active Maintenance contract will not be in a position to upgrade to the latest version of Chocolatey Licensed Extension.
```

Reviewer: Again just maintenance. An active support contract is not needed.

This can cause problems when there is a dependency that exists between a new version of Chocolatey CLI and the Chocolatey Licensed Extension.

> :memo: **NOTE:**
>
> This happened with the release of v1.0.0 of Chocolatey CLI, and this requirement was mentioned in the [release notes](xref:choco-release-notes#march-21-2022).
> We then released a [v1.0.1](xref:choco-release-notes#march-24-2022) release of Chocolatey with a fix (which was carried forward into [v1.1.0](xref:choco-release-notes#march-30-2022)) to allow Chocolatey CLI to work with the latest version of Chocolatey Licensed Extension, but this is no guarantee that this fix will be applied in future versions.

If you are in this position, we suggest that take one of the actions to prevent any unexpected changes to how Chocolatey CLI functions:

1. Request a Maintenance and Support contract renewal, upgrade the Chocolatey Licensed Extension to the latest  version and then upgrade Chocolatey CLI.

> 1. Forego the Maintenance and Support contract renewal and stay with a with the version of Chocolatey CLI you use. To prevent accidental upgrade of Chocolatey CLI we would suggest that you pin the version and ensure you are using an internal repository.

Author: Makes sense to me.

> 1. Forego the Maintenance and Support contract renewal and stay with a with the version of Chocolatey CLI you use. To prevent accidental upgrade of Chocolatey CLI we would suggest that you pin the version and ensure you are using an internal repository.

Author: Will get this updated

> - Forego the Maintenance and Support contract renewal and stay with a with the version of Chocolatey CLI and Chocolatey Licensed Extension that you use. To prevent accidental upgrade of Chocolatey CLI we would suggest that you [pin](xref:choco-command-pin) to the version you use and ensure you are using an [internal repository](xref:organizational-deployment-guide).

Reviewer: ```suggestion
- Forego the Maintenance and Support contract renewal and stay with the version of Chocolatey CLI and Chocolatey Licensed Extension that you use. To prevent accidental upgrade of Chocolatey CLI we would suggest that you [pin](xref:choco-command-pin) to the version you use and ensure you are using an [internal repository](xref:organizational-deployment-guide).
```

> ## Maintenance and Support contract considerations for perpetual license customers

Reviewer 2: ```suggestion
## Maintenance and Support contract considerations for Perpetual License customers
```

Having a Maintenance and Support contract with Chocolatey Software, Inc. consists of two parts:

> - Maintenance - allows you download newer versions of the licensed components of Chocolatey

Reviewer 2: ```suggestion
- Maintenance - allows you to download newer versions of the licensed components of Chocolatey.
```

> - Support - access to our commercial support team

Reviewer 2: ```suggestion
- Support - access to our commercial support team.
```

During the active period of the contract, you have access to both `Maintenance` and `Support`.
Once that expires, you would simply need to purchase that contract again to access one or both of those components.

> A perpetual license for Chocolatey means that you have a license which never expires.
> In addition, You would be able to use the Chocolatey Licensed Extension with the versions of the Chocolatey CLI that those licensed components were compatible with.
> And you would be able to download updates to the licensed components that came available during the active period of your Maintenance and Support contract.

Reviewer 2: ```suggestion
A Perpetual License means that you have a license which never expires and you are able to use the Chocolatey Licensed Extension with versions of Chocolatey CLI that are compatible.
While your Maintenance and Support contract is valid, you are able to download updates to the licensed components.
```

> If you are a perpetual license customer then you may not have an active Maintenance and Support contract, and therefore not be in a position to upgrade to the latest version of Chocolatey Licensed Extension.
> This can cause problems when there is a dependency that exists between a new version of Chocolatey CLI and the Chocolatey Licensed Extension.

Reviewer 2: ```suggestion
A Perpetual License customer that does not have a valid Maintenance and Support contract will not be in a position to upgrade to the latest version of Chocolatey Licensed Extension.
This can cause problems when there is a dependency that exists between a new version of Chocolatey CLI and the Chocolatey Licensed Extension.
```

> :memo: **NOTE:**
>
> This happened with the release of v1.0.0 of Chocolatey CLI, and this requirement was mentioned in the [release notes](xref:choco-release-notes#march-21-2022).
> We then released a [v1.0.1](xref:choco-release-notes#march-24-2022) release of Chocolatey with a fix (which was carried forward into [v1.1.0](xref:choco-release-notes#march-30-2022)) to allow Chocolatey CLI to work with the latest version of Chocolatey Licensed Extension, but this is no guarantee that this fix will be applied in future versions.

> If you are in this position, we would suggest that you follow one of the following two options to prevent any unexpected changes to how Chocolatey CLI functions:

Reviewer 2: ```suggestion
If you are in this position, we suggest that take one of the actions to prevent any unexpected changes to how Chocolatey CLI functions:
```

> 1. Request a Maintenance and Support renewal and upgrade the Chocolatey Licensed Extension to the newest version, then upgrade Chocolatey CLI.

Reviewer 2: ```suggestion
1. Request a Maintenance and Support contract renewal, upgrade the Chocolatey Licensed Extension to the latest  version and then upgrade Chocolatey CLI.
```

> 1. Forego the Maintenance and Support renewal and stick with an older version of Chocolatey CLI. Pin to that version and ensure you are using an internal repository and not linked up to the Chocolatey Community Repository directly for packaging updates.

Reviewer 2: ```suggestion
1. Forego the Maintenance and Support contract renewal and stay with a with the version of Chocolatey CLI you use. To prevent accidental upgrade of Chocolatey CLI we would suggest that you pin the version and ensure you are using an internal repository.
```

Reviewer 2: If we ask people to pin to a version of Chocolatey CLI, why do they need to ensure they are using an internal repository?

Author: @pauby this was the wording that was provided elsewhere, I just lifted it into this section.

> If you have any questions about anything related to the above, then please [reach out to the Sales team](https://chocolatey.org/contact/sales-general).

Reviewer 2: ```suggestion
If you have any questions about your Perpetual License or your Maintenance and Support contract, please [reach out to the Sales team](https://chocolatey.org/contact/sales-general).
```
